cmd/query/app/server_test.go

```diff
@@ -821,5 +857,7 @@ func TestServerHTTPTenancy(t *testing.T) {
 			}
 		})
 	}
+
 	server.Close()
+	wg.Wait()
```

Comment:
I think it is necessary, otherwise we can't guarantee all the server goroutines have shut down. Would you prefer it if `server.Close()` _blocked_ until all goroutines had shutdown?

Reply:
yes

Reply:
Address in this thread: https://github.com/jaegertracing/jaeger/pull/5299#discussion_r1540221532